src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -226,14 +232,15 @@ public override void Compile(string outputFile)
                 using (PerfEventSource.StartStopEvents.EmittingEvents())
                 {
                     NodeFactory.SetMarkingComplete();
-                    ReadyToRunObjectWriter.EmitObject(inputPeReader, outputFile, nodes, NodeFactory);
+                    ReadyToRunObjectWriter.EmitObject(inputPeReader, _outputFilePath, nodes, NodeFactory);
                 }
             }
         }
 
-        public override void WriteDependencyLog(string outputFileName)
+        public override void WriteDependencyLog()
         {
-            using (FileStream dgmlOutput = new FileStream(outputFileName, FileMode.Create))
+            string dgmlFileName = Path.ChangeExtension(_outputFilePath, ".dgml");
```

Comment:
The .map file being generated next to the executable already feels a bit wrong. It's the kind of file we should be generating into the intermediate directory, not into output path. Before we put more stuff on the same plan - do we know how this will fit into the .targets file?

Reply:
While most compilers I know typically generate auxiliary outputs like map files next to the executable, I think you have a valid point that this often ends up as busywork emitting everything under intermediates and subsequently copying just the executable to the output. What about some -auxoutpath option? Other than that, dgml is optional but I think that the map file as such is today always generated even though I doubt anyone actually uses it these days so we might also want to make that optional at some point.